gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/security/ApiKeyPlanBasedAuthenticationHandler.java

```diff
@@ -42,11 +42,17 @@ public ApiKeyPlanBasedAuthenticationHandler(AuthenticationHandler handler, Plan
     @Override
     protected boolean preCheckSubscription(AuthenticationContext authenticationContext) {
         if (!authenticationContext.contains(APIKEY_CONTEXT_ATTRIBUTE)) {
-            // There is no apikey at all, so no subscription to pre check.
+            // There is no apikey at all, so no subscription to pre-check.
             return true;
         }
+
         Optional<ApiKey> optApikey = (Optional<ApiKey>) authenticationContext.get(APIKEY_CONTEXT_ATTRIBUTE);
-        if (optApikey.isEmpty() || !optApikey.get().getPlan().equals(plan.getId())) {
+        if (optApikey.isEmpty()) {
```

Comment:
⚠️ I think that behavior also exists for v4 and for v4 with v2 definition support (aka jupiter).It must be aligned to avoid BC.